src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1650,11 +1649,43 @@ format layout_optimizer::get_preferred_format(program_node& node) {
         if (input_layout.format.dimension() == 5 &&
             (input_layout.data_type == data_types::f32 || input_layout.data_type == data_types::f16))
             expected = format::bfzyx;
+    } else if (node.is_type<resample>()) {
+        // if the resample is in the last part of the network and there are no users using blocked format,
+        // it is better to reorder to bfyx before resample is done.
+        if (all_users_simple_format(node, node, 0, 10)) {
+            const auto& dim = format::dimension(node.get_output_layout().format);
+            switch (dim) {
+                case 4 : expected = format::bfyx; break;
+                case 5 : expected = format::bfzyx; break;
+                case 6 : expected = format::bfwzyx; break;
+                default : throw std::runtime_error("Unknown rank " + dim);
+            }
+        } else {
+            expected = format::any;
+        }
     }
 
     return expected;
 }
 
+bool layout_optimizer::all_users_simple_format(program_node& origin_node, program_node& cur_node, int32_t cur_depth, int32_t max_depth) {
+    if (cur_node.is_output()) return true;
+    if (cur_depth > max_depth) return false;
```

Comment:
Is this condition correct? Even if all previous users have a simple format, it will return false on the last iteration and finally the function will return false, only because of this condition

Reply:
Yes, the intention is to fix the preferred format of resample only when the users are simple between the target resample and output layer.

Reply:
So, if this returns false, the format will be selected as any as before. (maybe I need to modify the name of this function to reduce confusion)

Reply:
Oh, I got it, thanks
Yeah, maybe it makes sense to slightly modify the name of function or add the description

Reply:
Oh, I got it, thanks
Yeah, maybe it makes sense to slightly modify the name of function or add the description

Reply:
Fixed the func name